ui/tests/acceptance/oidc-config/assignments-test.js

```diff
@@ -0,0 +1,197 @@
+    assert
+      .dom('[data-test-oidc-assignment-linked-block]')
+      .includesText('allow_all', 'displays default assignment');
+    assert.dom('[data-test-oidc-assignment-linked-block]').hasClass('is-disabled', 'Allow all is disabled.');
+  });
+
+  test('it creates, updates, and deletes an assignment', async function (assert) {
+    assert.expect(12);
+    await visit(ASSIGNMENTS_URL);
+    // create a new assignment
+    await click(SELECTORS.assignmentCreateButton);
+    assert.equal(
+      currentRouteName(),
+      'vault.cluster.access.oidc.assignments.create',
+      'navigates to create form in empty state'
+    );
+    await fillIn('[data-test-input="name"]', 'test-assignment');
+    await click('[data-test-component="search-select"]#entities .ember-basic-dropdown-trigger');
+    await click('.ember-power-select-option');
+    await click(SELECTORS.assignmentSaveButton);
+    assert.equal(
+      flashMessage.latestMessage,
+      'Successfully created the OIDC assignment test-assignment.',
+      'renders success flash upon creating the assignment'
+    );
+    assert.equal(
+      currentRouteName(),
+      'vault.cluster.access.oidc.assignments.assignment.details',
+      'navigates to the assignments detail view after save'
+    );
+    // assert default values in assignment details view are correct
+    assert.dom('[data-test-value-div="Name"]').hasText('test-assignment');
+    assert.dom('[data-test-value-div="Entities"]').hasText('1234-12345', 'shows the entity id.');
+
+    // edit assignment
+    await click(SELECTORS.assignmentEditButton);
+    assert.equal(
+      currentRouteName(),
+      'vault.cluster.access.oidc.assignments.assignment.edit',
+      'navigates to the assignment edit page from details'
+    );
+    await click('[data-test-component="search-select"]#groups .ember-basic-dropdown-trigger');
+    await click('.ember-power-select-option');
+    assert.dom('[data-test-oidc-assignment-save]').hasText('Update');
+    await click(SELECTORS.assignmentSaveButton);
+    assert.equal(
+      flashMessage.latestMessage,
+      'Successfully updated the OIDC assignment test-assignment.',
+      'renders success flash upon updating the assignment'
+    );
+
+    assert.dom('[data-test-value-div="Entities"]').hasText('1234-12345', 'it still shows the entity id.');
+    assert.dom('[data-test-value-div="Groups"]').hasText('abcdef-123', 'shows updated group name id.');
+
+    // delete the assignment
+    await click(SELECTORS.assignmentDeleteButton);
+    await click(SELECTORS.confirmDeleteButton);
+    assert.equal(
+      flashMessage.latestMessage,
+      'Assignment deleted successfully',
+      'renders success flash upon deleting assignment'
+    );
+    assert.equal(
+      currentRouteName(),
+      'vault.cluster.access.oidc.assignments.index',
+      'navigates back to assignment list view after delete'
+    );
+  });
+
+  test('it renders assignment list when assignments exist', async function (assert) {
```

Comment:
since there's no state difference between assignments existing and not existing this test may be more accurately named "it navigates to and from an assignment from the list view"